libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinPlugin.kt

```diff
@@ -326,15 +333,13 @@ internal class KotlinCommonSourceSetProcessor(
             project.tasks.findByName(kotlinCompilation.target.artifactsTaskName)?.dependsOn(kotlinTask)
```

Comment:
findByName triggers all task creations. There is no direct equivalent, named is the closest but might fail, so you would have to introduce a try catch.

Reply:
We have an equivalent convenience function called `locateTask` that wraps `try`-`named`-`catch`. We'll need to use it here.

Reply:
Already fixed in https://github.com/JetBrains/kotlin/pull/2996/files

@stephanenicolas Don't you want to review my PR instead?
Or maybe @h0tk3y could merge it into his branch so we see my edits here?

Reply:
@LouisCAD Sorry, I had not understood. I am gonna review your PR now.

Reply:
Feel free to PR my branch, I'll review it and merge any improvements.

Reply:
@stephanenicolas Feel free to PR my branch, I'll merge any improvement.

Reply:
@LouisCAD , it would take me a while to get started. I have built once only the kotlin repo, and never ran it. Sorry, it might look like a very lazy way to collaborate on my side by just adding comments. I am trying my best to follow up on a few bugs/issues, not only this one. Reviews is the best I can do for now. I wanna get to the point where I can build kotlin and try it, but I am not there yet.

Reply:
@stephanenicolas I did not build Kotlin. I only made edits using the IDE and Gradle docs for configuration avoidance.

I'm letting the Kotlin team do the checks, and publish dev or eap or milestone builds for me to try.